### PR TITLE
Update gambit and gerbil to new upstream releases

### PR DESCRIPTION
**Pull request by fare**
###### Motivation for this change

Update gambit and gerbil to new upstream releases

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
@GrahamcOfBorg build gambit gerbil

File: pkgs/development/compilers/gerbil/unstable.nix [hunk 1]
```diff
@@ -1,13 +1,13 @@
 { stdenv, callPackage, fetchgit, gambit-unstable }:
 
 callPackage ./build.nix {
-  version = "unstable-2018-09-06";
-  git-version = "0.14-DEV";
+  version = "unstable-2018-11-19";
+  git-version = "0.15-DEV";
   GAMBIT = gambit-unstable;
   SRC = fetchgit {
```
Comment (jtojnar) [inline pkgs/development/compilers/gerbil/unstable.nix]: Why not use `fetchFromGitHub`?

File: pkgs/development/compilers/gerbil/default.nix [hunk 2]
```diff
@@ -1,12 +1,12 @@
 { stdenv, callPackage, fetchurl, gambit }:
 
 callPackage ./build.nix {
-  version = "0.13";
-  git-version = "0.13";
+  version = "0.14";
+  git-version = "0.14";
   GAMBIT = gambit;
   SRC = fetchurl {
```
Comment (jtojnar) [inline pkgs/development/compilers/gerbil/default.nix]: Why not use `fetchFromGitHub`?

File: pkgs/development/compilers/gerbil/unstable.nix (hunk 1 above)
Comment (fare) [reply pkgs/development/compilers/gerbil/unstable.nix]: Let me try again that way...

Comment (fare) [reply pkgs/development/compilers/gerbil/unstable.nix]: It builds, so I amended things your way. done.

File: pkgs/development/compilers/gerbil/default.nix (hunk 2 above)
Comment (fare) [reply pkgs/development/compilers/gerbil/default.nix]: done.

File: pkgs/development/compilers/gambit/unstable.nix [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
-  SRC = fetchgit {
-    url = "https://github.com/feeley/gambit.git";
-    rev = "7cdc7e7b9194b2c088c0667efaf7686a4ffd0d8a";
-    sha256 = "06mmi8jkinihfirz4gjfw2lhxhskiqf3d47sihzx10r60asyqcxg";
+  version = "unstable-2018-11-16";
+# git-version = "4.9.1";
+  SRC = fetchFromGitHub {
+    owner = "feeley";
+    repo = "gambit";
+    rev = "79f5bf774e89c1220bece667598877821ca94b7c";
```
Comment (Mic92) [inline pkgs/development/compilers/gambit/unstable.nix]:
```suggestion
    version = "4.9.1";
    rev = "v${version}";
```

File: pkgs/development/compilers/gambit/unstable.nix [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
 callPackage ./build.nix {
-  version = "unstable-2018-09-03";
-# git-version = "4.9.0";
-  SRC = fetchgit {
-    url = "https://github.com/feeley/gambit.git";
-    rev = "7cdc7e7b9194b2c088c0667efaf7686a4ffd0d8a";
-    sha256 = "06mmi8jkinihfirz4gjfw2lhxhskiqf3d47sihzx10r60asyqcxg";
+  version = "unstable-2018-11-16";
+# git-version = "4.9.1";
+  SRC = fetchFromGitHub {
```
Comment (Mic92) [inline pkgs/development/compilers/gambit/unstable.nix]: this also needs to be adapted in build.nix:

```suggestion
  src = fetchFromGitHub {
```

File: pkgs/development/compilers/gerbil/default.nix (hunk 2 above)
Comment (Mic92) [reply pkgs/development/compilers/gerbil/default.nix]: this is still fetchurl

File: pkgs/development/compilers/gerbil/unstable.nix [hunk 5]
```diff
@@ -1,13 +1,14 @@
...
+  git-version = "0.15-DEV";
   GAMBIT = gambit-unstable;
-  SRC = fetchgit {
-    url = "https://github.com/vyzo/gerbil.git";
-    rev = "184cb635c82517d5d75d7966dcdf1d25ad863dac";
-    sha256 = "1ljzbpc36i9zpzfwra5hpfbgzj1dyzzp50h5jf976n8qr9x4l7an";
+  SRC = fetchFromGitHub {
+    owner = "vyzo";
+    repo = "gerbil";
+    rev = "11819599207c718ed50ac6f22bfa006a6c33ff4b";
```
Comment (Mic92) [inline pkgs/development/compilers/gerbil/unstable.nix]:
```suggestion
    rev = git-version;
```

**Review by Mic92 — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/gambit/unstable.nix (hunk 3 above)
Comment (Mic92) [reply pkgs/development/compilers/gambit/unstable.nix]: This can be also become part of the package name instead of having a date.

File: pkgs/development/compilers/gambit/bootstrap.nix [hunk 6]
```diff
@@ -1,16 +1,16 @@
...
 
   src = fetchurl {
-    url    = "http://www.iro.umontreal.ca/~gambit/download/gambit/v4.8/source/gambit-${tarball_version}-devel.tgz";
-    sha256 = "b7f86c794711792ca556ce41f8bc7043dffc395c01bb6d8d119bc2f454f89fbf";
+    url    = "http://www.iro.umontreal.ca/~gambit/download/gambit/v4.9/source/gambit-${tarball_version}-devel.tgz";
+    sha256 = "10kzv568gimp9nzh5xw0h01vw50wi68z3awfp9ibqrpq2l0n7mw7";
   };
 
-  buildInputs = [ autoconf ];
+  buildInputs = [ autoconf git ];
```
Comment (Mic92) [inline pkgs/development/compilers/gambit/bootstrap.nix]: How does it make use of git?

Comment (fare) [reply pkgs/development/compilers/gambit/bootstrap.nix]: Somewhere in the Makefile it uses git. Maybe not anymore, that we use fetchFromGitHub instead of fetchgit: that might be predicated on directory `.git` existing.

Comment (fare) [reply pkgs/development/compilers/gambit/bootstrap.nix]: Indeed the `lib/makefile` has a rule `deselect-gen-for-commit-post` that executes:
```
if [ -e ../.git ]; then git update-index --skip-worktree _kernel.c _system.c _num.c _std.c _eval.c _io.c _nonstd.c _thread.c _repl.c _gambit.c; fi
```
and the bootstrap tarball does contain the entire git tree (I'll ask the upstream author whether that's on purpose).

File: pkgs/development/compilers/gerbil/default.nix (hunk 2 above)
Comment (fare) [reply pkgs/development/compilers/gerbil/default.nix]: Oh, I had done it for unstable, but not default. Ok, done for default (in my next update, as soon as it builds).

File: pkgs/development/compilers/gambit/unstable.nix (hunk 3 above)
Comment (fare) [reply pkgs/development/compilers/gambit/unstable.nix]: Uh? I was told that for unstable, the date was the standard convention for the package version.

Also, as modifications are made after release, the unstable will track the master branch.

File: pkgs/development/compilers/gambit/unstable.nix (hunk 4 above)
Comment (fare) [reply pkgs/development/compilers/gambit/unstable.nix]: The reason I use `SRC` instead of `src` is that I failed to find the proper syntax in `build.nix` for the `std.mkDerivation` to use the `src` from the environment. I have:
```
{ stdenv, …, SRC } :
std.mkDerivation rec {
   src = SRC;
…}
```
If I tried to use `src = src;` then the `rec` understands that as an attempt to circularly define `src` as itself.

Comment (jtojnar) [reply pkgs/development/compilers/gambit/unstable.nix]: `inherit src;`

File: pkgs/development/compilers/gerbil/unstable.nix (hunk 5 above)
Comment (fare) [reply pkgs/development/compilers/gerbil/unstable.nix]: In general the revision does not correspond to an exact namable git version.

File: pkgs/development/compilers/gambit/unstable.nix (hunk 4 above)
Comment (fare) [reply pkgs/development/compilers/gambit/unstable.nix]: Thanks! I will update the PR after testing is complete.

**Comment by fare** [comment]
@GrahamcOfBorg build gambit gerbil gambit-unstable gerbil-unstable

File: pkgs/development/compilers/gambit/unstable.nix (hunk 3 above)
Comment (Mic92) [reply pkgs/development/compilers/gambit/unstable.nix]: Yes, we do this if we don't have a version number.
However this is not the case here, right?

File: pkgs/development/compilers/gambit/bootstrap.nix (hunk 6 above)
Comment (Mic92) [reply pkgs/development/compilers/gambit/bootstrap.nix]: Also `fetchgit` does not contain a `.git` directory.

Comment (Mic92) [reply pkgs/development/compilers/gambit/bootstrap.nix]: Can we remove it then?

**Comment by Mic92** [comment]
@GrahamcOfBorg build gambit gerbil gambit-unstable gerbil-unstable

File: pkgs/development/compilers/gambit/bootstrap.nix (hunk 6 above)
Comment (fare) [reply pkgs/development/compilers/gambit/bootstrap.nix]: For the bootstrap image, I use the recommended upstream tarball, that contains a `.git` directory and requires the `git` executable.

I could use `fetchgit` or `fetchFromGitHub` instead if you believe that's a better approach, and then remove the dependency on `git`.

File: pkgs/development/compilers/gambit/unstable.nix (hunk 3 above)
Comment (fare) [reply pkgs/development/compilers/gambit/unstable.nix]: It's generated automatically by an update script that tracks the git branch `master`. I just re-ran it and already now there is no simple version number usable the way you suggest.

File: pkgs/development/compilers/gambit/bootstrap.nix (hunk 6 above)
Comment (Mic92) [reply pkgs/development/compilers/gambit/bootstrap.nix]: ok. we can leave it then as is. `git` should be in `nativeBuildInputs` then.

Comment (fare) [reply pkgs/development/compilers/gambit/bootstrap.nix]: Oh, you mean put both of them in `nativeBuildInputs` instead of `buildInputs` ?

Comment (fare) [reply pkgs/development/compilers/gambit/bootstrap.nix]: Or should `git` be in `nativeBuildInputs` and `autoconf` be in `depsBuildInputs` ? I admit I don't understand the criteria for which inputs is which. Furthermore, I believe that cross-compilation of these particular packages may not be working, anyway.

Comment (fare) [reply pkgs/development/compilers/gambit/bootstrap.nix]: Is there a flag I should set somehow to tell nix that a cross-build is probably doomed to fail?

Comment (Mic92) [reply pkgs/development/compilers/gambit/bootstrap.nix]: It is ok, when the current expression does not support it, but it looks like the project itself support it.

Comment (fare) [reply pkgs/development/compilers/gambit/bootstrap.nix]: My current expression definitely doesn't support cross-compilation. I'm not even sure where to start from to support it, or how to test it. Probably with x86 vs x86_64, but even then I'm not sure how to do it.

File: pkgs/development/compilers/gerbil/build.nix [hunk 7]
```diff
@@ -1,20 +1,20 @@
...
   name    = "gerbil-${version}";
-  src     = SRC;
+  inherit src;
 
   # Use makeStaticLibraries to enable creation of statically linked binaries
   buildInputs_libraries = [ openssl zlib sqlite libxml2 libyaml mysql.connector-c lmdb leveldb postgresql ];
   buildInputs_staticLibraries = map makeStaticLibraries buildInputs_libraries;
 
-  buildInputs = [ GAMBIT coreutils rsync bash ]
+  buildInputs = [ gambit coreutils rsync bash ]
```
[the remark this answers is not in the document]
Comment (fare) [reply pkgs/development/compilers/gerbil/build.nix]: You mean, it's implicitly there so I can delete the explicit mention?

[the remark this answers is not in the document]
Comment (fare) [reply pkgs/development/compilers/gerbil/build.nix]: OK, I removed it and the package still builds. Thanks!

File: pkgs/development/compilers/gambit/bootstrap.nix (hunk 6 above)
Comment (fare) [reply pkgs/development/compilers/gambit/bootstrap.nix]: I'll mark as resolved: I don't currently have the time or skills to work on a recipe that supports cross-compilation, and I don't believe anyone else has the interest.

**Comment by fare** [comment]
@GrahamcOfBorg build gambit gerbil gambit-unstable gerbil-unstable

**Comment by 7c6f434c** [comment]
@GrahamcOfBorg build gambit gerbil gambit-unstable gerbil-unstable

File: pkgs/development/compilers/gambit/bootstrap.nix (hunk 6 above)
Comment (Mic92) [reply pkgs/development/compilers/gambit/bootstrap.nix]: We are trying to move it at least new code in a direction that makes it easier to reason about cross compilation at some point.
